### PR TITLE
Add support for Python 3.11

### PR DESCRIPTION
**Pull request by fmhoeger**
# CHANGELOG

- Infrastructure:
  - Added Python 3.11 to list of supported programming languages and to CI workflow
  - Updated `Dockerfile` to use Python 3.11

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -113,7 +113,7 @@ jobs:
         run: |
           tox ${{ needs.check_mturk_changes.outputs.needs_mturk_tests == 'true' && '-- --mturkfull' || ''}}
           npm run test --coverage
-        if: matrix.python-version == 3.10
+        if: matrix.python-version == 3.10 || matrix.python-version == 3.11
       - name: Run Fast Tests Only
```
Comment (pmcharrison) [inline .github/workflows/ci.yml:117]: I think the idea is that we only run MTurk tests once, so I don't think we need it on both 3.10 and 3.11.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -125,10 +125,10 @@ jobs:
         run: |
           tox -e fast
           npm run test --coverage
-        if: matrix.python-version != 3.10
+        if: matrix.python-version != 3.10 && matrix.python-version != 3.11
```
Comment (pmcharrison) [inline .github/workflows/ci.yml:128]: See above - I think we'll want to run fast tests for 3.10

File: .github/workflows/ci.yml (hunk 1 above)
Comment (fmhoeger) [reply .github/workflows/ci.yml:117]: Running MTurk tests on Python 3.11 only now

File: .github/workflows/ci.yml (hunk 2 above)
Comment (fmhoeger) [reply .github/workflows/ci.yml]: Running fast tests for 3.10 now

**Review by pmcharrison — APPROVED** [review]

**Pull request merged**